### DISCUSSION TITLE
Kyuubi OAuth flow

### DISCUSSION THREAD

**Discussion by yaroslavambrozyak** [apache/kyuubi#1651] [Q&A]
Hi. 
Recently we have implemented LDAP authentication for Kyuubi, it works well but we don`t want an additional LDAP server in our ecosystem. We are interested in implementing OAuth flow so the users can use their business google account to access the system. So we would like to integrate Kyuubi with Google OAuth and wondering is this doable using Kyuubi custom authentication or other mechanisms provided by Kyuubi? 
Appreciate any help. Thanks

**Comment by yaooqinn** [comment]
not familiar with oauth, but I think it's good to support by community officially

**Reply by mmuru** [reply]
@yaooqinn: The OpenID based authentication (OAuth2) such as Github and Google,  is commonly used SSO in most of the organizations without creating/managing user accounts and integrating with other applications. It will simplify Kyuubi authentication and integration.

**Reply by yaooqinn** [reply]
looks great to have it

**Comment by xifeng** [comment]
@yaroslavambrozyak 

acquire token or just validate token in kyuubi?

**Reply by yaroslavambrozyak** [reply]
Acquire.

For validating case as I understand we just need to implement CUSTOM authentication, receive the token via interface parameters, and do some operations. But I am interesting in getting the token  in the background, I mean a user will only pass his email
